Document the Browser Terminal (#3684)
gitpod-io/website: gitpod/docs/references/ides-and-editors/browser-terminal.md

> ---

Reviewer: Could you add a short changelog entry at the same time, Filip? 😁

Author: Pushed the entry ✅

section: ides-and-editors
title: Terminal (Browser)
---

<script context="module">
  export const prerender = true;
  import Keybind from "$lib/components/keybind.svelte";
</script>

# Terminal (Browser)

> The Browser Terminal is currently in [Early Access](/docs/help/public-roadmap/release-cycle) · [Send feedback](https://github.com/gitpod-io/gitpod/issues/17724).

> The Browser Terminal is a standalone IDE option built with the same technologies powering VS Code's terminals.

Reviewer: * Do we need to mention the same tech, e.g. xterm? 
Also is mentioning VS Code here potentially confusing?
* Should it be "Browser Terminal" capitalised? 

```suggestion
Use a terminal running in your browser as an IDE option to open your Gitpod projects. Use the terminal for simple and quick edits, or as your full time dedicated editing environment for all workspaces that you open. 
```

![Browser Terminal](../../../static/images/docs/browser-terminal.png)

> ## Accessing tasks

Reviewer: ```suggestion
## Viewing workspace tasks
```

> If you have pre-defined tasks in your `.gitpod.yml`, they will not appear right away as separate terminals like in JetBrains IDEs or VS Code. You can interface with tasks using [`gp tasks`](/docs/references/gitpod-cli#tasks).

Reviewer: ```suggestion
If you have defined tasks in your `.gitpod.yml` you can view and access those tasks using the command [`gp tasks`](/docs/references/gitpod-cli#tasks). 

> **Note:** Layout configuration such as `openMode` (e.g. `split-right`) are not supported. 
```

> ## Accessing ports

Reviewer: ```suggestion
## Managing workspace ports
```

> Similar to tasks, you may want to open different ports' URLs. You can do that with [`gp preview`](/docs/references/gitpod-cli#preview).

Reviewer: ```suggestion
To see open workspace ports, run [`gp ports list`](/docs/references/gitpod-cli#ports).
```

> > **Note**:💡 `gp preview` and `gp preview --external` have the same effect with the Browser Terminal, because split-view is not available there.

Reviewer: ```suggestion
> **Note**:💡 `gp preview` and `gp preview --external` have the same effect with the Browser Terminal, because split-view is not supported in a browser terminal.
```

## FAQs

> ### How can I create more terminals at once?

Reviewer: ```suggestion
### Can I view multiple terminals at the same time? e.g. split windows.
```

> For multitasking with multiple terminals at the same time, we recommend using `tmux`, which enables tiling your main terminal into multiple ones and much more.

Reviewer: ```suggestion
To open multiple terminals at the same time, we recommend `tmux`, which enables tiling your main terminal into multiple ones and much more. See [Dotfiles](/docs/configure/user-settings/dotfiles) for instructions on how to install tooling across all of your Gitpod workspaces. 
```

> **Read more**:
> 
> [A beginner's guide to tmux | Red Hat](https://www.redhat.com/sysadmin/introduction-tmux-linux)

Reviewer: ```suggestion
See: [A beginner's guide to tmux | Red Hat](https://www.redhat.com/sysadmin/introduction-tmux-linux) for more.
```

## Managing workspace tasks

If you have defined tasks in your `.gitpod.yml` you can view and access those tasks using the command [`gp tasks`](/docs/references/gitpod-cli#tasks).

> **Note:** Layout configuration such as `openMode` (e.g. `split-right`) are not supported.

## Managing workspace ports

To see open workspace ports, run [`gp ports list`](/docs/references/gitpod-cli#ports).

> **Note**: `gp preview` and `gp preview --external` have the same effect with the Browser Terminal, because split-view is not supported in a browser terminal.

## FAQs

### Can I view multiple terminals at the same time? e.g. split windows.

To splitting your terminal into different panes, we recommend `tmux`, which enables tiling your main terminal into multiple ones and much more. See [Dotfiles](/docs/configure/user-settings/dotfiles) for instructions on how to install tooling across all of your Gitpod workspaces.

To actually create a new Terminal process, you can simply open a new tab with the same URL. This will create a new terminal process in the same workspace.

See: [A beginner's guide to tmux | Red Hat](https://www.redhat.com/sysadmin/introduction-tmux-linux) for more.

> ### Can I use different shells?

Reviewer: Let's do it in a follow up? Would be nice, but not sure necessarily if it should be in core docs. If we had a community contributions section that would be good, since it's currently managed under asifs personal account.
